src/shazamkit.cs

```diff
@@ -0,0 +1,250 @@
+		bool Add (SHSignature signature, SHMediaItem[] mediaItems, [NullAllowed] out NSError error);
+
+		[Export ("addCustomCatalogFromURL:error:")]
+		bool Add (NSUrl url, [NullAllowed] out NSError error);
+
+		[Export ("writeToURL:error:")]
+		bool WriteL (NSUrl url, [NullAllowed] out NSError error);
+
+		[Watch (8,0), TV (15,0), Mac (12,0), iOS (15,0)]
+		[Export ("new")]
+		[return: Release]
+		SHCustomCatalog New ();
+	}
+
+	[iOS (15,0), Mac (12,0), Watch (8,0), TV (15,0), MacCatalyst (15,0)]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface SHMatch : NSSecureCoding
+	{
+		[Export ("mediaItems", ArgumentSemantic.Strong)]
+		SHMatchedMediaItem[] MediaItems { get; }
+
+		[Export ("querySignature", ArgumentSemantic.Strong)]
+		SHSignature QuerySignature { get; }
+	}
+
+	[iOS (15,0), Mac (12,0), Watch (8,0), TV (15,0), MacCatalyst (15,0)]
+	[BaseType (typeof (SHMediaItem))]
+	interface SHMatchedMediaItem : NSSecureCoding
+	{
+		[Export ("frequencySkew")]
+		float FrequencySkew { get; }
+
+		[Export ("matchOffset")]
+		double MatchOffset { get; }
+
+		[Export ("predictedCurrentMatchOffset")]
+		double PredictedCurrentMatchOffset { get; }
+	}
+
+	[iOS (15,0), Mac (12,0), Watch (8,0), TV (15,0), MacCatalyst (15,0)]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface SHMediaItem : NSSecureCoding, NSCopying
+	{
+		[NullAllowed, Export ("shazamID")]
+		string ShazamId { get; }
+
+		[NullAllowed, Export ("title")]
+		string Title { get; }
+
+		[NullAllowed, Export ("subtitle")]
+		string Subtitle { get; }
+
+		[NullAllowed, Export ("artist")]
+		string Artist { get; }
+
+		[Export ("genres", ArgumentSemantic.Strong)]
+		string[] Genres { get; }
+
+		[NullAllowed, Export ("appleMusicID")]
+		string AppleMusicId { get; }
+
+		[NullAllowed, Export ("appleMusicURL", ArgumentSemantic.Strong)]
+		NSUrl AppleMusicUrl { get; }
+
+		[NullAllowed, Export ("webURL", ArgumentSemantic.Strong)]
+		NSUrl WebUrl { get; }
+
+		[NullAllowed, Export ("artworkURL", ArgumentSemantic.Strong)]
+		NSUrl ArtworkUrl { get; }
+
+		[NullAllowed, Export ("videoURL", ArgumentSemantic.Strong)]
+		NSUrl VideoUrl { get; }
+
+		[Export ("explicitContent")]
+		bool ExplicitContent { get; }
+
+		[NullAllowed, Export ("isrc")]
+		string Isrc { get; }
+
+		[Export ("mediaItemWithProperties:")]
+		SHMediaItem MediaItem (NSDictionary<NSString, NSObject> properties);
+
+		[Async]
+		[Export ("fetchMediaItemWithShazamID:completionHandler:")]
+		void FetchMediaItem (string shazamID, Action<SHMediaItem, NSError> completionHandler);
+
+		[Export ("valueForProperty:")]
+		NSObject Value (string property);
+
+		[Export ("objectForKeyedSubscript:")]
+		NSObject ObjectForKeyedSubscript (string key);
+	}
+
+	[iOS (15,0), Mac (12,0), Watch (8,0), TV (15,0), MacCatalyst (15,0)]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface SHMediaLibrary
+	{
+		[Export ("defaultLibrary", ArgumentSemantic.Strong)]
+		SHMediaLibrary DefaultLibrary { get; }
```

Comment:
missing `[Static]` ?
https://developer.apple.com/documentation/shazamkit/shmedialibrary/3747200-defaultlibrary?language=objc shows `class`

Reply:
This is what I am currently confused about.
I had static on these selectors:
```
SHMediaItem::fetchMediaItemWithShazamID:completionHandler:
SHMediaItem::mediaItemWithProperties:
SHMediaLibrary::defaultLibrary
SHSignature::signatureWithDataRepresentation:error:
```
but Intro on iOS complains and says that these should not actually be static.
So I removed them and add to the ignore file, but then the TV intro says that they should be static?

Reply:
Definitely static, what is the intro erro on iOS?

Reply:
Let me pull it up!

Reply:
![image](https://user-images.githubusercontent.com/50846373/126820627-aaba803d-4896-4eb6-8b7c-5f47600c73ba.png)

Reply:
Try using them in an ObjC app
They might just not be "responding" but might work at runtime
If they don't work file an issue with Apple and attach your sample

Reply:
So I tried these out [here](https://gist.github.com/tj-devel709/2911a893c5105c31b307900221ca657d) 
and it looks like SHCustomCatalog::new and SHMediaLibrary::defaultLibrary are working, but the rest are not recognized!
I'll create an issue with Apple

Reply:
Just to confirm, I should file it here? www.feedbackassistant.apple.com

Reply:
@tj-devel709 yes

Reply:
Okay, I have filed an issue with apple!

Reply:
If useful here, this is the sample I gave to Apple.
[ShazamStaticTest.zip](https://github.com/xamarin/xamarin-macios/files/6880406/ShazamStaticTest.zip)

Reply:
Should I make comments about the ones that are not behaving as static for now or wait until the next beta and see if the issue is resolved?

Reply:
b4 is out now :)
but, to get things rolling, you can always comment what's incorrect (or unsure) and file an issue (and add it to the checklist)

Reply:
I was very wrong. I just was not calling the methods the correct way in xcode!

Reply:
@tj-devel709 so the methods do work, correct? we should close the maccore issue you created then.